lib/elixir/lib/io.ex

```diff
@@ -415,8 +415,9 @@ defmodule IO do
   See `IO.getn/3` for a description of return values.
 
   """
-  @spec getn(chardata | String.Chars.t(), pos_integer) :: chardata | nodata
-  @spec getn(device, chardata | String.Chars.t()) :: chardata | nodata
```

Comment:
I don't think this one has overlapping domains as devices are either PIDs or atoms?

Reply:
It does becuase `String.Chars.t/0` is defined as term.